server/server.go

```diff
@@ -591,14 +620,19 @@ func (server HTTPServer) singleRuleEndpoint(writer http.ResponseWriter, request
 		return
 	}
 
-	rule, successful = server.fetchRuleContent(*aggregatorResponse)
+	rule, successful, _ = server.fetchRuleContent(*aggregatorResponse, server.getOSDFlag(request))
 
 	if !successful {
+		err := responses.SendNotFound(writer, "Rule was not found")
```

Comment:
You could use only handleServerError here https://github.com/RedHatInsights/insights-operator-utils/blob/master/types/errors.go#L91

Reply:
`singleRuleEndpoint` don't get ruleID and ruleError, so using extracting that params second time just for this error, I think it's don't worth it https://github.com/RedHatInsights/insights-operator-utils/blob/15514d9c64e25edf80fee81e229b74c4b70b27e5/types/errors.go#L110

Reply:
`aggregatorResponse.Module`, but ok, not a big deal